packages/codeeditor/src/widget.ts

```diff
@@ -141,7 +141,7 @@ export class CodeEditorWrapper extends Widget {
    * A message handler invoked on an `'after-show'` message.
    */
   protected onAfterShow(msg: Message): void {
-    if (this._updateOnShow) {
+    if (this._updateOnShow || !this._hasRefreshed) {
```

Comment:
It seems like this should be `&&`, not `||`? Also, should `_hasRefreshed` be reset to false on hide?

Reply:
I updated the variable name and added a comment, as well as a reset whenever we attach.

Reply:
Thanks. That name change and comment made the logic clearer. I'll test.